### PR TITLE
Group operators

### PR DESCRIPTION
**Pull request by diegodrf**
PR to solve the issue #1.

### CODE REVIEW

File: LinqMarbles.Client/Components/Layout/NavMenu.razor [hunk 1]
```diff
@@ -1,37 +1,59 @@
...
+    </MudNavGroup>
+    <MudNavGroup Title="Order" Expanded="@true">
+        <MudNavLink Href="/order">Order</MudNavLink>
+        <MudNavLink Href="/order-descending">OrderDescending</MudNavLink>
+    </MudNavGroup>
+    <MudNavGroup Title="Projection" Expanded="@true">
+        <MudNavLink Href="/select">Select</MudNavLink>
+        <MudNavLink Href="/select-many">SelectMany</MudNavLink>
+    </MudNavGroup>
+    <MudNavGroup Title="Qualification" Expanded="@true">
```
Comment (linkdotnet) [inline LinqMarbles.Client/Components/Layout/NavMenu.razor:44]: That should be "Quantification", not? (See: https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/quantifier-operations)
```suggestion
    <MudNavGroup Title="Quantification" Expanded="@true">
```

**Review by linkdotnet — CHANGES_REQUESTED** [review]
LGTM - just one litte change.
Thanks for the effort @diegodrf

File: LinqMarbles.Client/Components/Layout/NavMenu.razor (hunk 1 above)
Comment (diegodrf) [reply LinqMarbles.Client/Components/Layout/NavMenu.razor:44]: You are right. 
I will fix it.

**Comment by linkdotnet** [comment]
Another question I had: Do we want to open all groups from the get-go?
And: Do we want to sort Operator-Groups (and also operators) alphabetically?
My gut feeling tells me yes (they way you did it).

Any feelings for the first question @diegodrf ?

**Review by linkdotnet — APPROVED** [review]

**Comment by linkdotnet** [comment]
Closes #1

**Comment by diegodrf** [comment]
I had the same doubts about starting the page with everything expanded. 

To solve the target issue, I kept in mind just splitting into groups and keeping the same behaviour on the startup. But it may be a good idea to create a helper button to collapse and expand all.

Starting the page with all groups collapsed is visually better. But who will access the page? 
- A person that knows what are looking for? For example, I know that I need to use something to filter or I'm using `join` in my code and I just want to know how the `join` works.
- A person who doesn't know what are looking for and wants to explore all functions. For this person, keeping all groups expanded can attract attention to some items and help to explore the page.

Answering the question, I vote to start collapsing all. 

Additionally, what do you think about creating a new home page showing all groups using a grid layout and the name of each group in blocks? It will be mobile-friendly.

**Comment by linkdotnet** [comment]
I do agree - and we can tackle that in different ways:
1. The simplest solution would be adding a search bar on top of the navigation. With that, users who wanna know a specific operator can just look for it that way. Beginners / People who want to discover stuff can just click through.
2. If we have the homepage as you suggested, I totally agree with the "collapse all" behavior. For now, I would leave it like it is (everything is expanded).

That said - thanks again for your great help. Feel free to add the "main page" idea as a GitHub issue with your ideas. It is much appreciated. 

For now, I'd like to merge the PR if there is nothing open from your side that we could not tackle in a follow-up.

**Comment by diegodrf** [comment]
Sure. 
You can merge the PR.

**Pull request merged**